src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/event.py

```diff
@@ -116,18 +113,55 @@ def event(self, ifname, vhid, event):
         finally:
             self.middleware.call_sync('failover.status_refresh')
 
+    @private
+    def generate_failover_data(self):
+
+        data = {}
+
+        failovercfg = self.middleware.call_sync('failover.config')
+        pools = self.middleware.call_sync('pool.query')
+        interfaces = self.middleware.call_sync('interface.query')
+        internal_ints = self.middleware.call_sync('failover.internal_interfaces')
+
+        data = {
+            'disabled': failovercfg['disabled'],
+            'master': failovercfg['master'],
+            'timeout': failovercfg['timeout'],
+            'groups': defaultdict(list),
+            'volumes': [
+                i['name'] for i in filter_list(pools, [('encrypt', '<', 2)])
+            ],
+            'phrasedvolumes': [
+                i['name'] for i in filter_list(pools, [('encrypt', '=', 2)])
+            ],
+            'non_crit_interfaces': [
+                i['id'] for i in filter_list(interfaces, [
+                    ('failover_virtual_aliases', '!=', []),
+                    ('failover_critical', '=', True),
+                ])
+            ],
+            'internal_interfaces': internal_ints,
+        }
+
+        for i in filter_list(interfaces, [('failover_critical', '=', True)]):
+            data['groups'][i['failover_group']].append(i['id'])
+
+        return data
+
     @private
     def _event(self, ifname, vhid, event):
 
+        fobj = self.generate_failover_data()
+
+        if not fobj:
+            self.logger.warning('Unable to acquire failover information! Exiting.')
+            return
```

Comment:
fobj could never be false, so this is a no-op

Reply:
I've removed the check.